tests/Set-DbaPowerPlan.Tests.ps1

```diff
@@ -5,7 +5,7 @@ Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
 Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
     Context "Validate parameters" {
         [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object {$_ -notin ('whatif', 'confirm')}
-        [object[]]$knownParameters = 'ComputerName', 'Credential', 'PowerPlan', 'CustomPowerPlan', 'InputObject', 'EnableException'
```

Comment:
Would removing the InputObject remove the authentication as well? Are you outputting Credential in Get?

Reply:
Yes, Get- returns the credential it gets. I removed InputObject because I think it does not add any value.
`Get-DbaPowerPlan -SqlInstance XYZ | Set-DbaPowerPlan` does not make any sense, because I don't want to set the power plan I just have. This would just set the best practice plan as `Set-DbaPowerPlan -SqlInstance XYZ` would do with less code.
And `Test-DbaPowerPlan -SqlInstance XYZ | Set-DbaPowerPlan` does not make any sense as well.
So I can only think of piping in ComputerNames - and that works.

Reply:
Ideally, the following should be possible for reasons like grabbing the powerplan of one machine in a group and duplicating it throughout the rest of the group. `Get-DbaPowerPlan -ComputerName XYZ -Credential ad\blah | Set-DbaPowerPlan -ComputerName ABC, DEF, HIJ` (with the set using the same credential as get-dbapowerplan, probably.
 
`Test-DbaPowerPlan -ComputerName sql2012 | Set-DbaPowerPlan` is currently supported and breaking changes should be introduced only when required. Before we understood PowerShell as well as we do now, we used Test instead of Get so a lot of our presentations refer to it. I'd like to keep major breaking changes for dbatools 2.0.

Would ValueFromPipelineByPropertyName better suit your needs?

Reply:
So you want to use `Get-DbaPowerPlan -ComputerName XYZ -Credential ad\blah | Set-DbaPowerPlan -ComputerName ABC, DEF, HIJ` instead of `Set-DbaPowerPlan -ComputerName ABC, DEF, HIJ -Credential ad\blah -PowerPlan (Get-DbaPowerPlan -ComputerName XYZ -Credential ad\blah).PowerPlan`?

Reply:
But `Get-DbaPowerPlan` also returns ComputerName - and that should not be used from the pipeline. So we have to test what comes over the pipeline and build solid rules what to use. Not that easy...

Reply:
If you like my implementation on Test-DbaPowerPlan, then I can try to use the same way with Set-DbaPowerPlan.

Reply:
Now everything you mentioned here works, so I will click on Resolve conversation...